File: Capitulo_01/01_Getting_Started:_The_Easy_Problems/02_Easy/12503_:::_Robot_Instructions/Main.cpp
using namespace std;
#include <cstdio>
#include <cstring>

int main()
{
	int casos, n, inst[100], s, rta;
	char line[10];
	scanf("%d", &casos);
	
	while(casos--)
	{
		rta=0;
		scanf("%d", &n);
		for(int i=0; i<n; i++)
		{
			scanf("%s", line);
			
			puts(line);
			//printf("%d", s);
			/*
			if(strcmp("LEFT", line)==0) {inst[i]=-1; rta--;}
			else if(strcmp("RIGHT", line)==0) {inst[i]=1; rta++;}
			else
			{
				s--;
				inst[i]=inst[s];
				
				rta+=inst[i];
			}
			*/
		}
		puts(".........");
		//printf("%d\n", rta);
	}
	return 0;
}

File: Capitulo_01/01_Getting_Started:_The_Easy_Problems/02_Easy/12503_:::_Robot_Instructions/Main.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main
{
	public static void main(String[] args) throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		//-----------------------------------------------------------
		int casos=Integer.parseInt(br.readLine()), n, inst[]=new int[100], s, rta;
		
		for(String line; casos--!=0;)
		{
			rta=0;
			n=Integer.parseInt(br.readLine());
			for(int i=0; i<n; i++)
			{
				line=br.readLine();
				if(line.equals("LEFT")) {inst[i]=-1; rta--;}
				else if(line.equals("RIGHT")) {inst[i]=1; rta++;}
				else
				{
					s=Integer.parseInt(line.split(" ")[2])-1;
					inst[i]=inst[s];
					
					rta+=inst[i];
				}
			}
			System.out.println(rta);
		} 
		//-----------------------------------------------------------
		br.close();
	}
}
